olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Messages.h

```diff
@@ -0,0 +1,215 @@
+  /**
+   * @brief Specifies partition of this metadata content.
+   *
+   * @param value The partition id string. Maximum length is 500 characters.
+   */
+  void SetPartition(std::string value) { partition_ = std::move(value); }
+
+  /**
+   * @return The checksum of the content.
+   */
+  const boost::optional<std::string>& GetChecksum() const { return checksum_; }
+  /**
+   * @brief Specified checksum of this metadata content.
+   *
+   * The algorithm used to calculate the checksum
+   * is user specific. Algorithms that can be used are, for example, MD5 or
+   * SHA1. This is not a secure hash, it's used only to detect changes in
+   * content.
+   * @param value Checksum string.
+   */
+  void SetChecksum(std::string value) { checksum_ = std::move(value); }
+
+  /**
+   * @return The compressed size of the content in bytes. Applicable also when
+   * `Content-Encoding` is set to gzip when uploading and downloading data.
+   * @note This will be present only if the \ref GetDataHandle is not empty.
+   */
+  const boost::optional<int64_t>& GetCompressedDataSize() const {
+    return compressed_data_size_;
+  }
+  /**
+   * @brief Specifies the size of the content.
+   *
+   * @param value Size of the content in bytes.
+   */
+  void SetCompressedDataSize(int64_t value) { compressed_data_size_ = value; }
+
+  /**
+   * @return The nominal size of the content in bytes. When compression is
+   * enabled, this is the size of the uncompressed content.
+   * @note This will be present only if the \ref GetDataHandle is not empty.
+   */
+  const boost::optional<int64_t>& GetDataSize() const { return data_size_; }
+  /**
+   * @brief Specifies the size of the content.
+   *
+   * @param value Size of the content in bytes.
+   */
+  void SetDataSize(int64_t value) { data_size_ = value; }
+
+  /**
+   * @return The content published directly in the metadata and encoded in
+   * base64 representation.
+   * @note This will be present only if the message size is less than or equal
+   * to 1 MB.
+   */
+  const Data& GetData() const { return data_; }
+  /**
+   * @brief Specifies the data of this content.
+   *
+   * @param value Data of this content as vector of bytes.
+   */
+  void SetData(Data value) { data_ = std::move(value); }
+
+  /**
+   * @return The handle created when uploading the content. It is used to
+   * retrieve the content at a later stage. Data Handle Example:
+   * 1b2ca68f-d4a0-4379-8120-cd025640510c.
+   * @note This will be present only if the message size is less than or equal
+   * to 1 MB.
+   */
+  const boost::optional<std::string>& GetDataHandle() const {
+    return data_handle_;
+  }
+
+  /**
+   * @brief Specifies data handle of this content.
+   *
+   * @param value The data handle as a string.
+   */
+  void SetDataHandle(std::string value) { data_handle_ = std::move(value); }
+
+  /**
+   * @return The timestamp of the content, in milliseconds since the Unix epoch.
```

Comment:
Can we shorten this sentence a bit?

"The Unix timestamp of the content (in milliseconds)"